examples/extensions/add_site_links.py

```diff
@@ -0,0 +1,219 @@
+    extension_feed_item3.targeted_campaign.value = campaign_resource_name
+    extension_feed_item3.device = feed_item_target_device_enum.MOBILE
+    final_url3 = extension_feed_item3.sitelink_feed_item.final_urls.add()
+    final_url3.value = 'http://www.example.com/mobile/wifi'
+
+    extension_feed_item_operation4 = client.get_type(
+        'ExtensionFeedItemOperation', version='v2')
+    extension_feed_item4 = extension_feed_item_operation4.create
+    extension_feed_item4.extension_type = extension_type_enum.SITELINK
+    extension_feed_item4.sitelink_feed_item.link_text.value = 'Happy hours'
+    extension_feed_item4.targeted_campaign.value = campaign_resource_name
+    extension_feed_item4.device = feed_item_target_device_enum.MOBILE
+    final_url4 = extension_feed_item4.sitelink_feed_item.final_urls.add()
+    final_url4.value = 'http://www.example.com/happyhours'
+    _populate_ad_schedule(
+        extension_feed_item4.ad_schedules.add(), day_of_week_enum.MONDAY, 18,
+        minute_of_hour_enum.ZERO, 21, minute_of_hour_enum.ZERO)
+    _populate_ad_schedule(
+        extension_feed_item4.ad_schedules.add(), day_of_week_enum.TUESDAY, 18,
+        minute_of_hour_enum.ZERO, 21, minute_of_hour_enum.ZERO)
+    _populate_ad_schedule(
+        extension_feed_item4.ad_schedules.add(), day_of_week_enum.WEDNESDAY, 18,
+        minute_of_hour_enum.ZERO, 21, minute_of_hour_enum.ZERO)
+    _populate_ad_schedule(
+        extension_feed_item4.ad_schedules.add(), day_of_week_enum.THURSDAY, 18,
+        minute_of_hour_enum.ZERO, 21, minute_of_hour_enum.ZERO)
+    _populate_ad_schedule(
+        extension_feed_item4.ad_schedules.add(), day_of_week_enum.FRIDAY, 18,
+        minute_of_hour_enum.ZERO, 21, minute_of_hour_enum.ZERO)
+
+    # Add extension feed items
+    try:
+        feed_response = (
+            extension_feed_item_service.mutate_extension_feed_items(
+                customer_id, [extension_feed_item_operation1,
+                              extension_feed_item_operation2,
+                              extension_feed_item_operation3,
+                              extension_feed_item_operation4]))
+    except GoogleAdsException as ex:
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:')
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}"')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f'\t\tOn field: {field_path_element.field_name}')
+        sys.exit(1)
+
+    print('Created ExtensionFeedItems:')
+    for feed_item in feed_response.results:
+        print(f'\tResource name: {feed_item.resource_name}')
+
+    return [_to_string_value(client, resource_name) for resource_name in
+            [result.resource_name for result in feed_response.results]]
```

Comment:
nested list comprehensions aren't very legible

Reply:
True, perhaps we shouldn't use those in an example.